.github/workflows/bindings-rust.yml

```diff
@@ -9,72 +9,147 @@ on:
-      uses: actions-rs/toolchain@v1
-      with:
-        toolchain: nightly
-        components: rustfmt, clippy
-
-    - name: Rustfmt
-      run: |
-        cd bindings/rust/
-        cargo +nightly fmt --all -- --check
-
-    - name: Test on standalone crate
-      run: |
-        cd bindings/rust/
-        export WASMEDGE_DIR="$(pwd)/../../"
-        cargo test --lib --examples
-        cargo test --doc
-
-    - name: Clean up
-      run: |
-        cd bindings/rust/
-        cargo clean
-
-    - name: Build WasmEdge using clang with Debug mode
-      run: |
-        cmake -Bbuild -GNinja -DCMAKE_BUILD_TYPE=Debug -DWASMEDGE_BUILD_SHARED_LIB=ON .
-        cmake --build build
-
-    - name: Clippy
-      run: |
-        cd bindings/rust/
-        export WASMEDGE_DIR="$(pwd)/../../"
-        export WASMEDGE_BUILD_DIR="$(pwd)/../../build"
-        cargo clippy --profile test --lib --examples -- -D warnings -D clippy::dbg_macro
-
-    - name: Test
-      run: |
-        cd bindings/rust/
-        export WASMEDGE_DIR="$(pwd)/../../"
-        export WASMEDGE_BUILD_DIR="$(pwd)/../../build"
-        export LD_LIBRARY_PATH="$(pwd)/../../build/lib/api"
-        cargo test --lib --examples --locked
-        cargo test --doc --locked
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Install Stable Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          components: rustfmt, clippy
+
+      - name: Install Nightly Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
+          components: rustfmt, clippy
+
+      - name: Rustfmt
+        run: |
+          cd bindings/rust/
+          cargo +nightly fmt --all -- --check
+
+      - name: Test on standalone crate
+        run: |
+          cd bindings/rust/
+          export WASMEDGE_DIR="$(pwd)/../../"
+          cargo test --lib --examples
+          cargo test --doc
+
+      - name: Clean up
+        run: |
+          cd bindings/rust/
+          cargo clean
+
+      - name: Build WasmEdge using clang with Debug mode
+        run: |
+          cmake -Bbuild -GNinja -DCMAKE_BUILD_TYPE=Debug -DWASMEDGE_BUILD_SHARED_LIB=ON .
+          cmake --build build
+
+      - name: Clippy
+        run: |
+          cd bindings/rust/
+          export WASMEDGE_DIR="$(pwd)/../../"
+          export WASMEDGE_BUILD_DIR="$(pwd)/../../build"
+          cargo clippy --profile test --lib --examples -- -D warnings -D clippy::dbg_macro
+
+      - name: Test
+        run: |
+          cd bindings/rust/
+          export WASMEDGE_DIR="$(pwd)/../../"
+          export WASMEDGE_BUILD_DIR="$(pwd)/../../build"
+          export LD_LIBRARY_PATH="$(pwd)/../../build/lib/api"
+          cargo test --lib --examples --locked
+          cargo test --doc --locked
+
+  build_macos:
+    name: MacOS
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [macos-11]
+    needs: get_version_v2
```

Comment:
Not sure why do you need this?

Reply:
According to the description from [GitHub](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners), `macos-latest` or `macos-11` denotes `macOS Big Sur 11` currently. When a new version of MacOS is released in the future, for example, macos-12, `macos-latest` cannot cover macos-11 at that time.

Reply:
I mean the `get_version_v2`. This is useless.

Reply:
By using [macos-11] (or [macos-11, macos-11] in the future),  I think we can have a much finer-grained control on the OS version. Thanks!

Reply:
I am fine with the `macos-11` part.
However, looks like there is a misunderstanding. What does this line `needs: get_version_v2` mean?

Reply:
> 

Sorry, I thought the comment was for line 91. I'll remove it, thanks a lot!